src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java

```diff
@@ -121,18 +122,23 @@ public Future<Void> updatePieceRecord(Piece piece, RequestContext requestContext
       .compose(v -> inventoryManager.updateItemWithPieceFields(piece, requestContext))
       .onSuccess(vVoid ->
         pieceStorageService.getPieceById(piece.getId(), requestContext).onSuccess(pieceStorage -> {
+            Piece.ReceivingStatus receivingStatusUpdate = piece.getReceivingStatus();
             Piece.ReceivingStatus receivingStatusStorage = pieceStorage.getReceivingStatus();
+            boolean isReceivingStatusChanged = receivingStatusStorage.compareTo(receivingStatusUpdate) != 0;
+
+            if(isReceivingStatusChanged) {
```

Comment:
let's add info log here that can help us in troubleshooting

Reply:
we already have loggin statements in #L138 and #L139

Reply:
basically they are with debug level and basically separate log that include pieceId and both statuses(from and to) would be useful expessionally if we have logic with conditional statements, but it up to you